File: bank-app/app/src/main/java/com/android/bankapp/login/router/LoginRouter.java
package com.android.bankapp.login.router;

import android.content.Context;

import com.android.bankapp.login.view.LoginActivity;
import com.android.bankapp.statements.StatementsActivity_;

import java.lang.ref.WeakReference;

public class LoginRouter implements LoginRouterInput {
    public WeakReference<LoginActivity> activity;

    @Override
    public void goToNextScreen(Context context) {
        StatementsActivity_.intent(context).start();
    }
}

File: app/src/main/java/com/earaujo/santander/login/LoginRouter.kt
package com.earaujo.santander.login

import android.content.Intent
import com.earaujo.santander.statements.StatementsActivity
import java.lang.ref.WeakReference

class LoginRouter : LoginRouterInput {

    var activity: WeakReference<LoginActivity>? = null

    fun passDataToNextScene(intent: Intent) {
        activity?.get()?.let {
            val userData = it.userAccount
            intent.putExtra(StatementsActivity.INTENT_USER_DATA, userData)
        }
    }

    override fun startStatementScreen(loginModel: LoginActivityModel) {
        activity?.get()?.let {
            val intent = Intent(it, StatementsActivity::class.java)
            passDataToNextScene(intent)
            it.startActivity(intent)
        }
    }

}

interface LoginRouterInput {
    fun startStatementScreen(loginModel: LoginActivityModel)
}